src/main/java/gregtech/api/capability/impl/BoilerRecipeLogic.java

```diff
@@ -0,0 +1,282 @@
+            if (denseFuelRecipe != null) {
+                // should exit here
+            }
+        }
+*/
+        if (!didStartRecipe) {
+            IItemHandlerModifiable importItems = boiler.getImportItems();
+            for (int i = 0; i < importItems.getSlots(); i++) {
+                ItemStack stack = importItems.getStackInSlot(i);
+                int fuelBurnTime = (int) Math.ceil(ModHandler.getFuelValue(stack));
+                if (fuelBurnTime / 80 > 0) {
+                    this.excessFuel += fuelBurnTime % 80;
+                    int excessProgress = this.excessFuel / 80;
+                    setMaxProgress(excessProgress + adjustBurnTimeForConfig(boiler.boilerType.runtimeBoost(fuelBurnTime / 80)));
+                    this.progressTime = 1;
+                    this.recipeEUt = adjustEUtForConfig(boiler.boilerType.steamPerTick());
+
+                    // avoid some NPEs
+                    this.fluidOutputs = Collections.emptyList();
+                    this.itemOutputs = NonNullList.create();
+
+                    stack.shrink(1);
+
+                    if (wasActiveAndNeedsUpdate) {
+                        wasActiveAndNeedsUpdate = false;
+                    } else {
+                        setActive(true);
+                    }
+                    break;
+                }
+            }
+        }
+        metaTileEntity.getNotifiedItemInputList().clear();
+        metaTileEntity.getNotifiedFluidInputList().clear();
+    }
+
+    @Override
+    protected void updateRecipeProgress() {
+        // todo figure out how maintenance fits in
+        int generatedSteam = this.recipeEUt * this.currentHeat / getMaximumHeat();
+        if (generatedSteam > 0) {
+            long amount = (generatedSteam + STEAM_PER_WATER) / STEAM_PER_WATER;
+            excessWater += amount * STEAM_PER_WATER - generatedSteam;
+            amount -= excessWater / STEAM_PER_WATER;
+            excessWater %= STEAM_PER_WATER;
+
+            FluidStack drainedWater = ModHandler.getWaterFromContainer(getInputTank(), (int) amount, true);
+            if (amount != 0 && (drainedWater == null || drainedWater.amount < amount)) {
+                getMetaTileEntity().explodeMultiblock();
+            } else {
+                setLastTickSteam(generatedSteam);
+                getOutputTank().fill(ModHandler.getSteam(generatedSteam), true);
+            }
+        }
+        if (currentHeat < getMaximumHeat()) {
+            setHeat(currentHeat + 1);
+        }
+
+        if (++progressTime > maxProgressTime) {
+            completeRecipe();
+        }
+    }
+
+    private int adjustEUtForConfig(int rawEUt) {
+        int throttle = ((MetaTileEntityLargeBoiler) metaTileEntity).getThrottle();
+        return Math.max(25, (int) (rawEUt * (throttle / 100.0)));
+    }
+
+    private int adjustBurnTimeForConfig(int rawBurnTime) {
```

Comment:
This does not seem to have anything to do with config values, so maybe it should be renamed.